pkg/controller/csi-driver-lvm/actuator.go

```diff
@@ -0,0 +1,807 @@
+								},
+							},
+						},
+						{
+							Name: "registration-dir",
+							VolumeSource: corev1.VolumeSource{
+								HostPath: &corev1.HostPathVolumeSource{
+									Path: "/var/lib/kubelet/plugins_registry",
+									Type: &hostPathTypeDir,
+								},
+							},
+						},
+						{
+							Name: "plugins-dir",
+							VolumeSource: corev1.VolumeSource{
+								HostPath: &corev1.HostPathVolumeSource{
+									Path: "/var/lib/kubelet/plugins",
+									Type: &hostPathTypeDir,
+								},
+							},
+						},
+						{
+							Name: "dev-dir",
+							VolumeSource: corev1.VolumeSource{
+								HostPath: &corev1.HostPathVolumeSource{
+									Path: "/dev",
+									Type: &hostPathTypeDir,
+								},
+							},
+						},
+						{
+							Name: "mod-dir",
+							VolumeSource: corev1.VolumeSource{
+								HostPath: &corev1.HostPathVolumeSource{
+									Path: "/lib/modules",
+								},
+							},
+						},
+						{
+							Name: "lvmcache",
+							VolumeSource: corev1.VolumeSource{
+								HostPath: &corev1.HostPathVolumeSource{
+									Path: pointer.SafeDeref(csidriverlvmConfig.HostWritePath) + "/cache",
+									Type: &hostPathTypeCreate,
+								},
+							},
+						},
+						{
+							Name: "lvmarchive",
+							VolumeSource: corev1.VolumeSource{
+								HostPath: &corev1.HostPathVolumeSource{
+									Path: pointer.SafeDeref(csidriverlvmConfig.HostWritePath) + "/archive",
+									Type: &hostPathTypeCreate,
+								},
+							},
+						},
+						{
+							Name: "lvmbackup",
+							VolumeSource: corev1.VolumeSource{
+								HostPath: &corev1.HostPathVolumeSource{
+									Path: pointer.SafeDeref(csidriverlvmConfig.HostWritePath) + "/backup",
+									Type: &hostPathTypeCreate,
+								},
+							},
+						},
+						{
+							Name: "lvmlock",
+							VolumeSource: corev1.VolumeSource{
+								HostPath: &corev1.HostPathVolumeSource{
+									Path: pointer.SafeDeref(csidriverlvmConfig.HostWritePath) + "/lock",
+									Type: &hostPathTypeCreate,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	objects := []client.Object{
+		csidriverlvmDriver,
+		csidriverlvmServiceAccountPlugin,
+		csidriverlvmClusterRolePlugin,
+		csidriverlvmClusterRoleBindingPlugin,
+		csidriverlvmDefaultStorageClass,
+		csidriverlvmLinearStorageClass,
+		csidriverlvmMirrorStorageClass,
+		csidriverlvmStripedStorageClass,
+		csidriverlvmDaemonSetPlugin,
+	}
+
+	return objects, nil
+}
+
+func (a *actuator) isOldCsiLvmExisting(ctx context.Context, shootNamespace string) (bool, error) {
+	_, shootClient, err := gutil.NewClientForShoot(ctx, a.client, shootNamespace, client.Options{}, extensionsconfig.RESTOptions{})
+
+	if err != nil {
+		return true, fmt.Errorf("failed to create shoot client: %w", err)
+	}
+
+	namespace := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: oldNamespace,
+		},
+	}
+
+	err = shootClient.Get(ctx, client.ObjectKeyFromObject(namespace), namespace)
```

Comment:
Is there a specific reason why not just listing the storage classes and checking for the existence of an old provisioner name?